geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/ConvertToGeoFile.scala

```diff
@@ -0,0 +1,238 @@
+/***********************************************************************
+ * Copyright (c) 2015-2021 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.geomesa.nifi.datastore.processor
+
+import java.io.{File, OutputStream}
+import java.nio.file.Files
+import java.util.UUID
+import java.util.zip.GZIPOutputStream
+
+import org.apache.commons.io.FilenameUtils
+import org.apache.nifi.annotation.behavior.{WritesAttribute, WritesAttributes}
+import org.apache.nifi.annotation.documentation.{CapabilityDescription, Tags}
+import org.apache.nifi.components.PropertyDescriptor
+import org.apache.nifi.expression.ExpressionLanguageScope
+import org.apache.nifi.flowfile.FlowFile
+import org.apache.nifi.processor._
+import org.apache.nifi.processor.io.OutputStreamCallback
+import org.apache.nifi.processor.util.StandardValidators
+import org.geomesa.nifi.datastore.processor.mixins.ConvertInputProcessor
+import org.geomesa.nifi.datastore.processor.mixins.ConvertInputProcessor.ConverterCallback
+import org.geomesa.nifi.datastore.processor.validators.GzipLevelValidator
+import org.geotools.util.factory.Hints
+import org.locationtech.geomesa.index.conf.QueryHints
+import org.locationtech.geomesa.tools.`export`.formats.FeatureExporter.ExportStream
+import org.locationtech.geomesa.tools.`export`.formats.FileSystemExporter.{OrcFileSystemExporter, ParquetFileSystemExporter}
+import org.locationtech.geomesa.tools.`export`.formats._
+import org.locationtech.geomesa.utils.io.{CloseWithLogging, PathUtils}
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+import scala.util.control.NonFatal
+
+@Tags(Array("OGC", "geo", "convert", "converter", "simple feature", "geotools", "geomesa",
+  "parquet", "orc", "arrow", "gml", "geojson", "csv", "leaflet"))
```

Comment:
Leaflet.  ROFL...  <sigh>

Reply:
lol, i could remove it if you want. it wasn't any extra work so i left it in

Reply:
Yeah, it was just funny.;)